spec/controllers/api/user_sessions_controller_spec.rb

```diff
@@ -93,4 +118,86 @@
       end
     end
   end
+
+  private
+
+  def create_session!(attributes)
+    Session.create!(
+      title: attributes.fetch(:title, "Example Session"),
+      user: attributes.fetch(:user),
+      uuid: attributes.fetch(:uuid, "uuid"),
+      start_time: DateTime.current,
+      start_time_local: DateTime.current,
+      end_time: DateTime.current,
+      end_time_local: DateTime.current,
+      type: "MobileSession",
+      longitude: 1.0,
+      latitude: 1.0,
+      is_indoor: false
+    )
+  end
+
+  def create_stream!(attributes)
+    Stream.create!(
+      sensor_package_name: "AirBeam2:00189610719F",
+      sensor_name: "AirBeam2-F",
+      measurement_type: "Temperature",
+      unit_name: "Fahrenheit",
+      session: attributes.fetch(:session),
+      measurement_short_type: "dB",
+      unit_symbol: "dB",
+      threshold_very_low: 20,
+      threshold_low: 60,
+      threshold_medium: 70,
+      threshold_high: 80,
+      threshold_very_high: 100
+    )
+  end
+
+  def create_measurements!(attributes)
+    Measurement.create!(
+      time: Time.current,
+      latitude: 1,
+      longitude: 1,
+      value: 1,
+      milliseconds: 0,
+      stream: attributes.fetch(:stream)
+    )
+  end
+
+  def session_data(attributes)
+    "[
+    {\"calibration\":0,
+      \"contribute\":true,
+      \"drawable\":2131165435,
+      \"end_time\":\"2019-05-24T11:37:19\",
+      \"is_indoor\":false,
+      \"latitude\":0.0,
+      \"location\":\"http://localhost/s/4oefo\",
+      \"longitude\":0.0,
+      \"deleted\":#{attributes.fetch(:deleted, false)},
+      \"notes\":[],
+      \"start_time\":\"2019-05-24T11:37:16\",
+      \"streams\":{\"Phone Microphone\"
+        :{\"average_value\":0.0,
+          \"deleted\":false,
+          \"measurement_type\":\"Sound Level\",
+          \"measurements\":[],
+          \"sensor_package_name\":\"Builtin\",
+          \"sensor_name\":\"Phone Microphone\",
+          \"measurement_short_type\":\"dB\",
+          \"unit_symbol\":\"dB\",
+          \"threshold_high\":80,
+          \"threshold_low\":60,
+          \"threshold_medium\":70,
+          \"threshold_very_high\":100,
+          \"threshold_very_low\":20,
+          \"unit_name\":\"decibels\"}
+        },
+      \"tag_list\":\"#{attributes.fetch(:tag_list, "")}\",
+      \"title\":\"#{attributes.fetch(:title, "title")}\",
+      \"type\":\"MobileSession\",
+      \"uuid\":\"#{attributes.fetch(:uuid, "uuid")}\"}
+    ]"
```

Comment:
I'd suggest something like
```
[{..}, ...].to_json
```
That way you should be able to avoid escaping and quotes